### DISCUSSION TITLE
training about multi-speaker vits

### DISCUSSION THREAD

**Discussion by Hemiptelea** [coqui-ai/TTS#2119] [General Q&A] [answered]
Hi. I'm new to TTS and I have questions.

I have public dataset(good quality, many files) I want to make a model based on this. 

And also I have private dataset(normal quality, 1 hour long)

There is no need to output the results in the voice of the public dataset. then, After initially learning as a public dataset single speaker then, Should I proceed with multi-speaker learning with a private dataset? 

Or do I conduct multi-speaker learning from the beginning? 

And we are preparing the dataset with various voices. Is it possible to add new speakers after training the model? If so, can you provide a detailed guide? Thanks for reading.

**Comment by shigabeev** [comment, marked as answer]
I would do the following:
1. Take model from model zoo as a pretrain
2. Combine your public dataset with private one into a single dataset
3. For speaker embedding, I would take average of speaker embeddings for each speaker, since you don't need to go beyond your training set

Let me know how it worked out.

**Reply by Hemiptelea** [reply]
What should I do if the model zoo does not have the language I want?

**Reply by shigabeev** [reply]
Models with different language would still act as a better starting point rather than random init.

**Reply by Hemiptelea** [reply]
Please check if my understanding is correct. 

First. All datasets are merged into one and trained at once. 

Second. After training, implement speaker_embeddings using compute_embeddings.py.

**Reply by huongtxvega** [reply]
@queechy Hi, Have you tried it yet?

**Reply by Hemiptelea** [reply]
I have succeeded in creating the desired language in other ways, but the above method is not used. @huongtxvega

**Reply by huongtx498** [reply]
@queechy I'm confused between finetune single speaker and multispeaker with speaker embedding, can u share how u do? Many thanks.

**Reply by Hemiptelea** [reply]
> @queechy I'm confused between finetune single speaker and multispeaker with speaker embedding, can u share how u do? Many thanks.

Can you explain me about what you are trying to do?

**Reply by huongtx498** [reply]
@queechy Same as your case, i am having 3 different voices reading 20h in total, but the voice i want to model only has 4h.

**Reply by Hemiptelea** [reply]
> @queechy Same as your case, i am having 3 different voices reading 20h in total, but the voice i want to model only has 4h. 

In my case, I used single-speaker way.

First I trained the model with good quality datasets
up to about 250,000 steps 

And I replaced dataset to my private dataset and decreased lr value

it sounds good, and I satisfied 

If you want multi-speaker way, you'd better find a another explanation 

sorry for my bad english

@trinhhuong244

**Reply by huongtx498** [reply]
@queechy no problem, how much lr value you reduced?

**Reply by Hemiptelea** [reply]
> @queechy no problem, how much lr value you reduced?

It was a long time ago so I don't remember very well 

I think it was around 0.0004. There is no set value, so try several values.

**Reply by huongtx498** [reply]
@queechy okay, i will try, thank you very much.
